File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/python/mouredev.py
import math


def check_prime_fibonacci_even(number):

    result = f"{number} "

    # Primo
    if number > 1:
        for index in range(2, number):
            if number % index == 0:
                result += "no es primo, "
                break
        else:
            result += "es primo, "

    else:
        result += "no es primo, "

    # Fibonacci
    result += "es fibonacci " if number > 0 and (is_perfect_square(5 * number * number + 4) or is_perfect_square(
        5 * number * number - 4)) else "no es fibonacci "

    # Par
    result += "y es par" if number % 2 == 0 else "y es impar"

    print(result)


def is_perfect_square(number):
    sqrt = int(math.sqrt(number))
    return sqrt * sqrt == number


check_prime_fibonacci_even(2)
check_prime_fibonacci_even(7)
check_prime_fibonacci_even(0)
check_prime_fibonacci_even(1)
check_prime_fibonacci_even(-2)

File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/kotlin/mouredev.kt

import kotlin.math.sqrt

fun main() {
    checkPrimeFibonacciEven(2)
    checkPrimeFibonacciEven(7)
    checkPrimeFibonacciEven(0)
    checkPrimeFibonacciEven(1)
    checkPrimeFibonacciEven(-2)
}

private fun checkPrimeFibonacciEven(number: Int) {

    var result = "$number "

    // Primo
    if (number > 1) {
        var prime = true
        for (index in 2 until number) {
            if (number % index == 0) {
                result += "no es primo, "
                prime = false
                break
            }
        }
        if (prime) {
            result += "es primo, "
        }
    } else {
        result += "no es primo, "
    }

    // Fibonacci
    result += if (number > 0 && (isPerfectSquare(5 * number * number + 4) || isPerfectSquare(
    5 * number * number - 4))) "es fibonacci " else "no es fibonacci "

    // Par
    result += if (number % 2 == 0) "y es par" else "y es impar"

    println(result)
}


private fun isPerfectSquare(number: Int): Boolean {
    val sqrt = sqrt(number.toDouble()).toInt()
    return sqrt * sqrt == number
}